src/message-builder.js

```diff
@@ -1,53 +1,112 @@
+  }
+}
+function stringToDateConverter(messages) {
+  for (let message of messages) {
+    message.post_at = new Date(message.post_at);
+    if (message.repeat) {
+      message.repeat = new Date(message.repeat);
     }
+  }
+  return messages;
 }
 
 function convertChannelNameToId(channel, channels) {
-    for (let c of channels) {
-        if (c.id == channel) {
-            return channel;
-        }
-        if (c.name == channel) {
-            return c.id;
-        }
+  for (let c of channels) {
+    if (c.id == channel) {
+      return channel;
+    }
+    if (c.name == channel) {
+      return c.id;
     }
-    return null;
+  }
+  return null;
 }
 
+function incrimentDateIterative(startDate, endDate) {
+  let date = new Date(startDate);
+  let dateList = [];
+
+  while (date < endDate) {
+    dateList.push(date);
+    date = new Date(startDate.setDate(startDate.getDate() + 7));
+  }
+
+  return dateList;
+}
+
+function buildRepeatMessages(messages) {
+  messages = stringToDateConverter(messages);
+  allMessages = [];
+
+  for (let i = 0; i < messages.length; i++) {
+    if (messages[i].repeat) {
+      const dates = incrimentDateIterative(
+        messages[i].post_at,
+        messages[i].repeat
+      );
+      //TODO figure out how to modify indv object properties, instead of rebuilding it
+      //! problem: modyfying indv object changed *all* object properties to be uniform
+      //*This also gets rid of 'repeat' proterty
+      for (let j = 0; j < dates.length; j++) {
+        allMessages.push({
+          text: messages[i].text,
+          post_at: dates[j],
+          channel: messages[i].channel,
+          user: messages[i].user,
+          file: messages[i].file,
+        });
+      }
+    } else {
+      allMessages.push(messages[i]);
+    }
+  }
+
+  return allMessages;
+}
 
 function areMessagesCorrect(messages, userChannels, users) {
-    users = Object.keys(users).concat(["default"])
-
-    for (let message of messages) {
-        checkField(message, "text")
-        checkField(message, "post_at")
-        checkField(message, "channel")
-
-        try {
-            Date.parse(message.post_at)
-        } catch (e) {
-            throw `${e} in file ${message.file}`;
-        }
-
-        // CHECK IF HAS USER || DEFAULT AND IF USER EXISTS        
-        const user = message.user || "default";
-        if (!users.includes(user)) {
-            throw `${user} is not found in token list in a message in file ${message.file}`
-        }
-        const channels = userChannels[user];
-
-        if (convertChannelNameToId(message.channel, channels) == null) {
-            throw `Channel ${message.channel} not found for message in file ${message.file}`;
-        }
+  users = Object.keys(users).concat(["default"]);
+
+  for (let message of messages) {
+    checkField(message, "text");
+    checkField(message, "post_at");
+    checkField(message, "channel");
+
+    try {
+      Date.parse(message.post_at);
```

Comment:
We also need to add here at post_until parsing

Reply:
are you refering to 'repeat'? the field does not exist anymore once it gets to this phase. The repeated messages are built in the buildRepeatMessages function, and then the 'repeat' property is deleted. 
So in the end, a list of all messages, including repeating, are prebuilt and then sent to the API. I was thinking this was the best approach because my understanding once it is posted to api, it cannot change. What do you think?
Ex: One message 'message' entered to be posted/repeated for three weeks consecutavily. {'message", post_at: date, repeat: endDate}
What is sent to the api is {"message", post_at: date},{"message", post_at: date2},{"message", post_at: endDate}

Reply:
Sorry, I think my intention for `areMessagesCorrect` are not clear from the code:

The aim of `areMessagesCorrect` is there to ensure that all messages and their respective fields defined by the user are in the correct format, have a user, and that the user can see the channel where the messages and to stop executing the action if not everything is correctly set up.

It also gives a clear error in which file the error occurs, allows for easy fixing.

Once this correctness is ensured, the action can proceed and the code won't be cluttered with extra error handling for user inputs. 

The truth is that the we can never rely on the fact that the message yaml-files are correctly and we need to check them first before doing any processing.

Example:
If the date for `repeat` is not properly formatted it would currently throw an error, but it does not say which yaml-file fucks it up. This makes fixing the error really hard.

Solution:
Add the checking to the `areMessagesCorrect` and move the `buildRepeatMessage` a bit down in code.

Reply:
I was testing the try/catch.. noticed that during `try{Date.parse()}` a date string that is not formatted correctly will not be caught as an error. It returns a NaN so does not get passed through to the catch logic. This then continues and the program runs with the error in the Date object.

Reply:
Good catch! You are right, `Date.parse()` does not throw an exeception but returns NaN. That definitly needs to be changed as well. (By me not by you :D)

I think you can just add a comment that it should be checked then, and move down the buildRepeatMessage Code. Once we have that we can merge, and I can properly rewrite the areMessagesCorrect function 👍